Fix zim file creation on Windows.

mgautierfr opened the pull request:
The bug comes from the fact that windows silently replaces `\n` by `\r\n` if we open the file "normally" (in text mode).

Fix #507

First commit is not necessary (windows replaces **silently**), but it is better to have it anyway.
(I've pushed the branch in two steps to have the CI run on the two last commits).

veloman-yunkan reviewed (commented):
It makes sense to fix `TempFile::fd()`, too: https://github.com/openzim/libzim/blob/1ed3b93313c714c9f9f32b87c64ccfffe53aa642/test/tools.cpp#L68-L78

mgautierfr commented:
I've also fixed `TempFile::fd`.

I've changed header unittest with the same data. Test doesn't fails because we write and read the data using the same fd (so with the same open mode). Windows replace `\n` by `\r\n` when writing but also change the data when reading. So we've got a coherent value.

Now we don't change the path of the zim file (and we reuse the path of the tempfile), the final path already exists and we must allow to replace an existing when we move the file at the end of the process.
